### PR TITLE
KAFKA-8644. The Kafka protocol generator should allow null defaults for bytes and array fields

### PR DESCRIPTION
**Pull request by cmccabe**

### CODE REVIEW

File: generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java [hunk 1]
```diff
@@ -1221,19 +1229,19 @@ private String fieldDefault(FieldSpec field) {
...
             }
         } else if (field.type().isBytes()) {
-            if (!field.defaultString().isEmpty()) {
+            if (field.defaultString().equals("null")) {
+                validateNullDefault(field);
+                return "null";
+            } else if (!field.defaultString().isEmpty()) {
                 throw new RuntimeException("Invalid default for bytes field " +
-                    field.name() + ": custom defaults are not supported for bytes fields.");
+                        field.name() + ".  The only valid default for an array field " +
```
Comment (stanislavkozlovski) [inline generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java]: typo: "for a byte field", right?

Comment (cmccabe) [reply generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java]: Good catch

**Review by stanislavkozlovski — APPROVED** [review]
LGTM.

**Comment by gwenshap** [comment]
LGTM. Thank you for the patch and the review. 
Merging.

**Pull request closed without merging**